### PR TITLE
HCL 2.0 rebuild

### LINKED ISSUE #9: Malformed table in the Github `repository` module
On https://registry.terraform.io/modules/withmethod/repository/github/0.4.0:

image

### LINKED ISSUE #10: Add archived attribute
Request for enhancement:

Add the [`archived`](https://www.terraform.io/docs/providers/github/d/repository.html#archived) attribute for the github_repository resource.

### LINKED ISSUE #12: Type conflict
req_status_checks_context is declared as a "list" but implemented as a string.  

```
Error: Incorrect attribute value type

  on .terraform/modules/this_github.this_repo/withmethod-terraform-github-repository-5aa2138/main.tf line 34, in resource "github_branch_protection" "protected-branch":
  34:     contexts = ["${var.req_status_checks_context}"]
```

### PR DESCRIPTION
**Pull request by ksatirli**
Hey folks,

following the rebuilding of other modules, I finally got around to updating this one, too.

`terraform-github-repository` holds a special place, as it is the first module I ever published to the Terraform Registry.

History aside, the changes in this PR are numerous:

0.) brought issue templates, Workflows and `LICENSE` in line with other modules
1.) updated all code to HCL 2.0
2.) removed work-arounds (or, more aptly: hacky hacks) and used cleaner implementations
3.) added support for a handlful of new / missing resources: 

- using template repository 
- creating issue labels 
- creating deploy keys 
- creating repository projects 
- creating repository files (released in GH Provider `2.4.x`

4.) heavily expanded outputs and added some that are needed for special situations
5.) added a large number of examples

This PR touches some thirty-ish files, but the ones that are of the highest importance for reviewing are `README.md`, `main.tf`, `variables.tf`, and `outputs.tf`.

### CODE REVIEW

**Review by bbaassssiiee — DISMISSED** [review]
You spent a great deal of time on this, and you improved a lot of the meta-bits. In testing you should pay attention to managing the branch protections. If possible share a mock organization with some repos that you manage with this, so the reviewers can see the changes applied.

**Comment by ksatirli** [comment]
Thank you for the review, @bbaassssiiee!

The @operatehappy happy organization is actually Terraformed using a development version of this module and I plan on publishing the `meta` repository that has all the configuration (and thus: implementation) in it.

Just need to find some time ... 😬

**Comment by ksatirli** [comment]
Hey @onlydole,

Your input for the other two modules is very much appreciated, so I'm hoping I can convince you to have a look at this module, too.

This one does a lot more, and (I think!) is ready for prime-time. It is a complete rewrite of the very first module I ever pushed to the [registry](https://registry.terraform.io/modules/withmethod/repository/github/0.4.0), taking into account all the beauty of `0.12`.

Thoughts?

File: README.md [hunk 1]
```diff
@@ -24,112 +33,105 @@ This module depends on a correctly configured [GitHub Provider](https://www.terr
...
-  req_pr_reviews_dismissal_users            = []
-  req_pr_reviews_dismissal_teams            = []
-  restrictions_users                        = []
-  restrictions_teams                        = []
+module "simple-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-simple-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline README.md]:
```suggestion
private = false
```

Private repositories are [free now](https://github.blog/2019-01-07-new-year-new-github/) in most contexts (OSS FTW) 🏆

File: examples/branch-protection/main.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module "branch-protection-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-complex-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/branch-protection/main.tf]: Nit: Same here

File: examples/files/main.tf [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module "files-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-files-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/files/main.tf]: Nit: Also here

File: examples/issue-labels/main.tf [hunk 4]
```diff
@@ -0,0 +1,25 @@
+module "issue-labels-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-issue-labels-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/issue-labels/main.tf]: Nit: also here

File: examples/project-columns/main.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
+module "project-column-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-project-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/project-columns/main.tf]: Nit: also here

File: examples/project/main.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+module "project-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-project-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/project/main.tf]: Nit: also here

File: examples/simple/main.tf [hunk 7]
```diff
@@ -0,0 +1,7 @@
+module "simple-example" {
+  source  = "operatehappy/repository/github"
+  version = "2.0.0"
+
+  name    = "oh-demo-simple-example"
+  private = false // NOTE: private repositories require a paid subscription
```
Comment (onlydole) [inline examples/simple/main.tf]: Nit: also here

**Review by onlydole — DISMISSED** [review]
VERY small nits - incredible overall! I can't wait to start using this module 🤩

File: README.md (hunk 1 above)
Comment (ksatirli) [reply README.md]: Oh my - you can tell how long I had this PR around 😂 - literally wrote this at the end of last year / beginning of this year and never got around to updating this.

Yes, VERY good catch. Thanks!

**Comment by ksatirli** [comment]
> VERY small nits - incredible overall! I can't wait to start using this module 🤩

Thanks for reviewing - I've removed the outdated note 🤦.

Excited to use this, too!

**Review by onlydole — APPROVED** [review]
LGTM 🥳

**Pull request merged**